central/xlsx/transform.go

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright (C) 2023 IOTech Ltd
+//
+
+package xlsx
+
+import (
+	"io"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+
+	"github.com/xuri/excelize/v2"
+)
+
+type mappingField struct {
+	defaultValue string // the default value defined in the MappingTable sheet
+	path         string // the path value defined in the MappingTable sheet
+}
+
+// ConvertXlsx transforms the xlsx file to the Converter interface
+func ConvertXlsx(file io.Reader, dto any) (Converter, error) {
+	f, err := excelize.OpenReader(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	fieldMappings, err := convertMappingTable(f)
+	if err != nil {
+		return nil, err
+	}
+
+	converter, err := processXlsxByType(f, fieldMappings, dto)
```

Comment:
if we create the convertor in the begining, we don't need `processXlsxByType` here
just executing the `convertToDTO` func here